apps/api/v2/src/ee/calendars/input/create-ics.input.ts

```diff
@@ -0,0 +1,52 @@
+import { ApiProperty } from "@nestjs/swagger";
+import {
+  ArrayNotEmpty,
+  IsBoolean,
+  IsOptional,
+  Validate,
+  ValidatorConstraint,
+  ValidatorConstraintInterface,
+} from "class-validator";
+import { IsNotEmpty, IsArray } from "class-validator";
+
+// Custom constraint to validate ICS URLs
+@ValidatorConstraint({ async: false })
+export class IsICSUrlConstraint implements ValidatorConstraintInterface {
+  validate(url: unknown) {
+    if (typeof url !== "string") return false;
+
+    // Check if it's a valid URL and ends with .ics
+    try {
+      const urlObject = new URL(url);
+      return (
+        urlObject.protocol === "http:" ||
+        (urlObject.protocol === "https:" && urlObject.pathname.endsWith(".ics"))
```

Comment:
If it's "http" protocol we are not checking if it ends with ".ics" but we should.